frontend/public/graphql/client.ts

```diff
@@ -1,92 +1,87 @@
+import { getK8sResourcePath } from '../module/k8s';
+import { K8sModel, K8sResourceCommon } from '../module/k8s/types';
+import { URLQuery } from './client.gql';
+import { URLQueryType, URLQueryVariables } from '../../@types/console/generated/graphql-schema';
+
 import { ApolloClient } from 'apollo-client';
 import { HttpLink } from 'apollo-link-http';
 import { SubscriptionClient } from 'subscriptions-transport-ws';
-import { InMemoryCache } from 'apollo-cache-inmemory';
+import { InMemoryCache, NormalizedCacheObject } from 'apollo-cache-inmemory';
 import { WebSocketLink } from 'apollo-link-ws';
 import { split } from 'apollo-link';
 
-import { getK8sResourcePath } from '../module/k8s';
-import { K8sKind, K8sResourceCommon } from '../module/k8s/types';
-import { URLQuery } from './client.gql';
-import { URLQueryType, URLQueryVariables } from '../../@types/console/generated/graphql-schema';
 import { getConsoleRequestHeaders, coFetch } from '../co-fetch';
+import {
+  HUB_CLUSTER_NAME,
+  LAST_CLUSTER_USER_SETTINGS_KEY,
+} from '@console/shared/src/constants/common';
+import { getValueFromStorage } from '@console/shared/src/utils/user-settings';
 
-let wssErrors = 0;
-
-class GraphQLReady {
-  private callback: VoidFunction;
-  private ready: boolean;
-  private wasCalled: boolean;
+export let client: ApolloClient<NormalizedCacheObject>;
+let subsClient: SubscriptionClient;
 
-  setReady() {
-    this.ready = true;
-    if (!this.wasCalled && this.callback) {
-      this.wasCalled = true;
-      this.callback();
-    }
-  }
+export const startGQLClient = (onReady?: VoidFunction) => {
+  subsClient?.close(true, false);
+  let wssErrors = 0;
+  const getActiveCluster = () =>
```

Comment:
Could we pass the cluster in as an arg? Technically, we are only using session storage to persist the active cluster across page reloads. If we ever changed that, it would break this client.

Reply:
yes, i agree. That will be much better